sec2_the-ruby-programming-language/2.11_oop_intro/3_getter.rb

```diff
@@ -0,0 +1,43 @@
+# Introduce Getter
+
+class Student
+  @first_name
+  @last_name
+  @email
+  @username
+  @password
+
+  def to_s
+    "First name: #{@first_name}"
+  end
+
+  # Method
+  def first_name(name)
+    @first_name = name
+  end
+
+  # Setter
+  def first_name=(name)
+    @first_name = name
+  end
+
+  # Getter
+  def first_name
+    @first_name
+  end
+end
+
+michelle = Student.new
+puts michelle                         # First name:         //(accessing the to_s built-in method)
+
+# (Refer line 42 to understand why line 34, 35 is commented out)
+# michelle.first_name("Michelle")
+# puts michelle                       # in `first_name': wrong number of arguments (given 1, expected 0) (ArgumentError)
+
+# Setter
+michelle.first_name = "Haha"
+puts michelle                         # First name: Haha
+
+# Getter
+# If need Getter, cannot have method
+puts michelle.first_name              # Haha
```

Comment:
1. Why when there is a getter, we cannot have the method (Refer Line 43, Line 34)
2. Is it possible to define setter without parameters? When usually do you need a setter without parameters?

Reply:
@tlkiong  how about this?

Reply:
@Michelle-Lohwt How about which? My comment below is to answer this as well.